holoviews/plotting/bokeh/chart.py

```diff
@@ -906,18 +903,36 @@ def get_data(self, element, ranges, style):
         hover = 'hover' in self.handles
 
         # Group by stack or group dim if necessary
+        xdiff = None
+        xvals = element.dimension_values(xdim)
         if group_dim is None:
             grouped = {0: element}
+            is_dt = isdatetime(xvals)
+            if is_dt or xvals.dtype.kind != 'O':
+                xdiff = np.diff(xvals)
+                if len(np.unique(xdiff)) == 1 and xdiff[0] == 0:
+                    xdiff = 1
+                if is_dt:
+                    width = xdiff.astype('timedelta64[ns]').astype(np.int64) * width / 1e6
+                else:
+                    width = width / xdiff
+                width = 1 - np.repeat(np.min(np.abs(width)), len(xvals))
+                data['width'] = [width]
+            else:
+                data['width'] = [np.repeat(width, len(xvals))]
+            width = 'width'
         else:
             grouped = element.groupby(group_dim, group_type=Dataset,
                                       container_type=dict,
                                       datatype=['dataframe', 'dictionary'])
+            data["width"] = [np.repeat(width, len(xvals))]
```

Comment:
If the width value is a constant it should simply be returned as part of the mapping, there's no point putting it in the data and duplicating it N times.